src/services/tx/tx-sender/sdk.ts

```diff
@@ -17,15 +18,30 @@ export const getAndValidateSafeSDK = (): Safe => {
   return safeSDK
 }
 
+export const switchWalletChain = (wallet: ConnectedWallet, chainId: SafeInfo['chainId']) => {
+  const chainIdInHex = ethers.utils.hexValue(BigNumber.from(chainId))
+
+  return wallet.provider.request({
```

Comment:
I wonder if this is in some way superior or inferior to `onboard.setChain`. Or perhaps it's the same thing.

Reply:
`setChain` indeed [makes this request](https://github.com/blocknative/web3-onboard/blob/develop/packages/core/src/chain.ts#L61) [internally](https://github.com/blocknative/web3-onboard/blob/develop/packages/core/src/provider.ts#L436-L444). Looks like there's a lot of validation beforehand which we may benefit from.

Reply:
As long as it's not affecting any internal state in onboard, I guess it's fine to bypass it. Passing a wallet arg is probably simpler than onboard.

Reply:
There is also some logic in place to handle hardware wallets (`handleChainSwitch`) where we have to first disconnect and reconnect again. It probably makes sense to reuse that function and rely on `onboard` directly.